File: spring/src/main/java/com/example/demo/entity/order/Order.java
package com.example.demo.entity.order;

import com.example.demo.entity.user.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "`order`")
public class Order {
    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne(optional = false)
    private User orderUser;

    @OneToMany(mappedBy = "order", cascade = {CascadeType.MERGE, CascadeType.PERSIST})
    private List<OrderProduct> orderList;

    public Order prepareSave() {
        this.orderList.forEach(orderProduct -> orderProduct.setOrder(this));
        return this;
    }
}

File: spring/src/main/kotlin/com/example/demo/entity/order/Order.kt
package com.example.demo.entity.order

import com.example.demo.entity.user.User
import javax.persistence.*

@Entity
@Table(name = "`order`")
data class Order (
    @Id
    @GeneratedValue
    val id: Long = 0,

    @OneToMany(mappedBy = "order", cascade = [CascadeType.MERGE, CascadeType.PERSIST])
    val orderList: List<OrderProduct> = mutableListOf(),

    @ManyToOne(optional = false)
    var orderUser: User
) {
    fun prepareSave(): Order {
        orderList.forEach {
            it.order = this
        }

        return this
    }
}